### ISSUE TITLE
Unable to test latest master

### ISSUE THREAD

**Issue by TheNeikos** [sebcrozet/ncollide#132]
```sh
neikos ~/p/r/ncollide (master)> cargo clean
neikos ~/p/r/ncollide (master)> git rev-parse HEAD
d97238c2b577a64b84776106a05adb5c4cf5bddb
neikos ~/p/r/ncollide (master)> git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working directory clean
neikos ~/p/r/ncollide (master)> cargo build
   Compiling rustc-serialize v0.3.19
   Compiling libc v0.2.11
   Compiling num-traits v0.1.32
   Compiling rand v0.3.14
   Compiling num-integer v0.1.32
   Compiling num-iter v0.1.32
   Compiling num-complex v0.1.32
   Compiling num-bigint v0.1.32
   Compiling num-rational v0.1.32
   Compiling num v0.1.32
   Compiling nalgebra v0.8.2
   Compiling ncollide_math v0.4.0 (file:///home/neikos/projects/rust/ncollide/ncollide_math)
   Compiling ncollide_utils v0.4.0 (file:///home/neikos/projects/rust/ncollide/ncollide_utils)
   Compiling ncollide_geometry v0.1.1 (file:///home/neikos/projects/rust/ncollide/ncollide_geometry)
   Compiling ncollide_procedural v0.4.0 (file:///home/neikos/projects/rust/ncollide/ncollide_procedural)
   Compiling ncollide_pipeline v0.8.1 (file:///home/neikos/projects/rust/ncollide/ncollide_pipeline)
   Compiling ncollide_transformation v0.4.0 (file:///home/neikos/projects/rust/ncollide/ncollide_transformation)
   Compiling ncollide v0.9.1 (file:///home/neikos/projects/rust/ncollide)
neikos ~/p/r/ncollide (master)> cargo test
   Compiling nom v1.2.2
   Compiling pkg-config v0.3.8
   Compiling lzw v0.9.0
   Compiling glfw-sys v3.1.3
   Compiling bitflags v0.3.3
   Compiling csfml-graphics-sys v0.1.0
   Compiling winapi-build v0.1.1
   Compiling khronos_api v1.0.0
   Compiling csfml-audio-sys v0.1.0
   Compiling byteorder v0.4.2
   Compiling csfml-network-sys v0.1.0
   Compiling kernel32-sys v0.2.2
   Compiling color_quant v1.0.0
   Compiling inflate v0.1.1
   Compiling bitflags v0.5.0
   Compiling winapi v0.2.6
   Compiling sfml-types v0.1.0
   Compiling csfml-window-sys v0.1.1
   Compiling glob v0.2.11
   Compiling gif v0.7.0
   Compiling csfml-system-sys v0.1.0
   Compiling freetype-sys v0.3.1
   Compiling time v0.1.35
   Compiling log v0.3.6
   Compiling semver v0.2.3
   Compiling enum_primitive v0.1.0
   Compiling ncollide_math v0.4.0
   Compiling xml-rs v0.2.2
   Compiling sfml v0.11.2
   Compiling ncollide_utils v0.4.0
   Compiling gcc v0.3.28
   Compiling miniz-sys v0.1.7
   Compiling libz-sys v1.0.3
   Compiling glfw v0.2.0
   Compiling flate2 v0.2.14
   Compiling freetype-rs v0.5.0
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302:9: 302:27 warning: lint raw_pointer_derive has been removed: using derive with raw pointers is ok
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302 #[allow(raw_pointer_derive)]
                                                                                                   ^~~~~~~~~~~~~~~~~~
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302:9: 302:27 warning: lint raw_pointer_derive has been removed: using derive with raw pointers is ok
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302 #[allow(raw_pointer_derive)]
                                                                                                   ^~~~~~~~~~~~~~~~~~
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:301:16: 301:21 note: in this expansion of #[derive_Clone] (defined in /home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs)
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302:9: 302:27 warning: lint raw_pointer_derive has been removed: using derive with raw pointers is ok
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:302 #[allow(raw_pointer_derive)]
                                                                                                   ^~~~~~~~~~~~~~~~~~
/home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs:301:10: 301:14 note: in this expansion of #[derive_Copy] (defined in /home/neikos/.cargo/registry/src/github.com-88ac128001ac3a9a/glfw-0.2.0/src/ffi/mod.rs)
   Compiling ncollide_procedural v0.4.0
   Compiling gl_generator v0.4.2
   Compiling ncollide_testbed2d v0.1.1 (file:///home/neikos/projects/rust/ncollide/ncollide_testbed2d)
   Compiling png v0.4.3
   Compiling gl v0.5.2
   Compiling image v0.6.1
   Compiling kiss3d v0.4.2
   Compiling ncollide_testbed3d v0.1.0 (file:///home/neikos/projects/rust/ncollide/ncollide_testbed3d)
ncollide_testbed3d/src/objects/convex.rs:27:44: 27:58 error: mismatched types:
 expected `ncollide_procedural::trimesh::TriMesh<na::structs::point::Point3<f32>>`,
    found `ncollide_procedural::trimesh::TriMesh<na::structs::point::Point3<f32>>`
(expected struct `ncollide_procedural::trimesh::TriMesh`,
    found a different struct `ncollide_procedural::trimesh::TriMesh`) [E0308]
ncollide_testbed3d/src/objects/convex.rs:27             gfx:        window.add_trimesh(convex.clone(), na::one())
                                                                                       ^~~~~~~~~~~~~~
ncollide_testbed3d/src/objects/convex.rs:27:44: 27:58 help: run `rustc --explain E0308` to see a detailed explanation
ncollide_testbed3d/src/objects/convex.rs:27:44: 27:58 note: Perhaps two different versions of crate `ncollide_procedural` are being used?
ncollide_testbed3d/src/objects/convex.rs:27             gfx:        window.add_trimesh(convex.clone(), na::one())
                                                                                       ^~~~~~~~~~~~~~
error: aborting due to previous error
Could not compile `ncollide_testbed3d`.

To learn more, run the command again with --verbose.
neikos ~/p/r/ncollide (master) [101]> 
```

If you check the compile output you will find that it compiles `ncollide_procedural` twice, once the local build, once one that it fetches from crates.io.

```
eikos ~/p/r/ncollide (master) [101]> cargo -V
cargo 0.10.0-nightly (d6e148d 2016-03-09)
neikos ~/p/r/ncollide (master)> rustc -V
rustc 1.9.0-nightly (c9629d61c 2016-03-10)
```

**Comment by dbrgn** [comment]
Same issue here.

**Comment by sebcrozet** [comment]
It looks like the example depends on the local `ncollide` while the testbed's renderer depends on the one from `crates.io`.

The only way I see to fix this is to move the `ncollide` examples using the testbeds into the testbeds' example folders (which seems more logical anyway).

**Comment by sebcrozet** [comment]
By the way @TheNeikos , if you attempt to compile the tests or examples from `ncollide` with the current nightly (or beta) version of `rustc`, you will end up with internal compiler errors. This will be fixed when rust-lang/rust#33478 lands. In the mean time you may use the stable channel.
